multiraft/storage.go

```diff
@@ -233,10 +233,17 @@ func (w *writeTask) start(stopper *stop.Stopper) {
 				groupResp := &groupWriteResponse{raftpb.HardState{}, -1, -1, groupReq.entries}
 				response.groups[groupID] = groupResp
 				if !raft.IsEmptySnap(groupReq.snapshot) {
-					err := group.ApplySnapshot(groupReq.snapshot)
-					if err != nil {
-						panic(err) // TODO(bdarnell)
-					}
+					gID := groupID
+					s.sendEvent(&EventSnapshot{
```

Comment:
@bdarnell sorry, I do not understand "remove ```snapshot``` from ```groupWriteRequest```". If remove ```snapshot```, where we get ```snapshot```?